File: app/src/main/java/com/lostincontext/awareness/AwarenessModule.java
package com.lostincontext.awareness;

import com.google.android.gms.awareness.Awareness;
import com.google.android.gms.common.api.GoogleApiClient;
import com.lostincontext.commons.BaseActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class AwarenessModule {

    private BaseActivity activity;

    public AwarenessModule(BaseActivity activity) {
        this.activity = activity;
    }

    @Provides
    GoogleApiClient provideGoogleApi() {
        return new GoogleApiClient.Builder(activity)
                .addApi(Awareness.API)
                .build();
    }

    @Provides
    BaseActivity provideBaseActivity() {
        return activity;
    }
}

File: app/src/main/java/com/lostincontext/awareness/AwarenessModule.kt
package com.lostincontext.awareness

import com.google.android.gms.awareness.Awareness
import com.google.android.gms.common.api.GoogleApiClient
import com.lostincontext.commons.BaseActivity

import dagger.Module
import dagger.Provides

@Module
class AwarenessModule(private val activity: BaseActivity) {

    @Provides
    internal fun provideGoogleApi(): GoogleApiClient {
        return GoogleApiClient.Builder(activity).addApi(Awareness.API).build()
    }

    @Provides
    internal fun provideBaseActivity(): BaseActivity {
        return activity
    }
}



